### PR TITLE
Add script for ROCm CI on Jenkins

### PR DESCRIPTION
**Pull request by kmaehashi**
We are about to introduce AMD GPU on Jenkins. This PR adds a script to run unittest on ROCm.

The CI script, `.pfnci/linux/run.sh`, defines each test *stage*. By combining multiple stages we can build a CI pipeline. This makes it possible to share test scripts on both Jenkins and FlexCI, so that we can later add CUDA tests to the script and migrate them from Jenkins to FlexCI.

Test stages:
- `build`: Builds a docker image used for testing.
- `push`: Push the built docker image to Google Container Registry (FlexCI).
- `cache_get`: Pull build/kernel cache from Google Cloud Storage (FlexCI).
- `cache_put`: Push build/kernel cache to Google Cloud Storage (FlexCI).
- `test`: Run tests.

CI pipeline on each test infra:
* Jenkins: `build`, `test`
* FlexCI (prep project): `build`, `push`
* FlexCI (test project): `cache_get`, `test`, `cache_put` (excluding pull-requests)

TODO:
- [x] implement a signal hander in the script to make tests terminatable

### CODE REVIEW

File: .pfnci/linux/run.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  if (( $# < 2 )); then
+    echo "${USAGE}"
+    exit 1;
+  fi
+
+  TARGET="$1"; shift
+  STAGES="$@"
+
+  repo_root="$(cd "$(dirname "${BASH_SOURCE}")/../.."; pwd)"
+  base_branch="$(cat "${repo_root}/.pfnci/BRANCH")"
```
Comment (kmaehashi) [inline .pfnci/linux/run.sh:41]: `BRANCH` file will be used to know the base branch of pull-requests.
By adding this file we no longer have to maintain the list of known branches like this:
https://github.com/cupy/cupy/blob/v10.0.0a2/.pfnci/script.sh#L154-L175

When backporting the content of the file should be changed to `v9`.

File: .pfnci/BRANCH [hunk 2]
```diff
@@ -0,0 +1 @@
+master
```
Comment (kmaehashi) [inline .pfnci/BRANCH:1]: When backporting the content of the file should be changed to `v9`.

**Comment by kmaehashi** [comment]
I'm now test driving the script here: https://jenkins.preferred.jp/job/chainer/job/cupy_master/

**Comment by kmaehashi** [comment]
The test ran fine. Let's introduce this to `cupy_pr` once we fix all the ROCm test failures.

/test

**Review by emcastillo — APPROVED** [review]

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**